### PR TITLE
Add workflow for automatically generating release

### PR DESCRIPTION
**Pull request by inahga**
Automates the Janus release process.

This workflow, when triggered by workflow_dispatch, will push a version bump commit to each active branch, then create a release on that new commit. That release will then trigger our other workflows, like publishing SQL schema and to crates.io.

There are a few tradeoffs to this approach:

This bypasses status checks. An operator should check the status of the main branch before using this workflow. I omitted an automated check for the latest status check because I don't want to end up in a situation where we have to make arbitrary commits to get the latest status check passing. I generally think that with per-PR status checks, it's very unlikely to release code that is flat out broken--likely blocking status checks would involve new clippy lints when stable is updated, or rustsec violations.

This workflow elegantly but also inelegantly handles prerelease branches. `cargo-edit` bumps version `0.7.0-prerelease-1` to `0.7.0`. This is great for when we're ready to release it for real, but not so great when we need incremental prerelease versions. I don't think there's a tidy solution to this, so we'll just have to handle incremental prerelease versions manually.

There is no facility for supplying extra release notes. I figure we can just edit the created release if we need to add more detail to a set of release notes.

Even with these tradeoffs, this workflow should make releasing Janus a 1-click process (modulo prerelease branches).

### Deploying

I will need @branlwyd's help with creating the `DIVVIUP_GITHUB_AUTOMATION_RELEASE_PAT` secret. We will want to make this an organization secret, so we can roll out this approach to multiple repositories. It will need `repo` scopes, and it *must* have an expiration date (otherwise repo cloning won't work).

We will also need to assign `divviup-github-automation` to the bypass list in the branch protection rules. This is denoted `Allow specified actors to bypass required pull requests`. I will do this if this PR is approved.

### CODE REVIEW

**Comment by inahga** [comment]
This workflow can be demo'd in https://github.com/divviup/inahga-test.

**Comment by inahga** [comment]
Closes https://github.com/divviup/janus/issues/293

**Review by divergentdave — APPROVED** [review]
I think this looks good, it'll take care of some purely mechanical steps

**Comment by branlwyd** [comment]
I generated a PAT & set the permissions appropriately (I think) -- all non-pictured perms are not checked:

image

This PAT is available via the `DIVVIUP_GITHUB_AUTOMATION_RELEASE_PAT` org secret.

File: .github/workflows/make-release.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      - name: Restore cargo-edit from cache
+        id: cache-cargo-edit-restore
+        uses: actions/cache/restore@v4
+        with:
+          path: |
+            ~/.cargo/bin
+          key: ${{ runner.os }}-cargo-edit
+
+      - run: cargo install cargo-edit --no-default-features --features set-version
+        if: steps.cache-cargo-edit-restore.outputs.cache-hit != 'true'
```
Comment (branlwyd) [inline .github/workflows/make-release.yml:40]: Barring cache eviction, what will allow us to upgrade to newer versions of `cargo-edit`? (If it's nothing, WDYT about just reinstalling `cargo-edit` each time -- at least on my machine, it only takes a few seconds to compile & install. This isn't such a good suggestion if the install takes much longer on GH's actions runners.)

**Review by branlwyd — APPROVED** [review]

File: .github/workflows/make-release.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+on:
+  workflow_dispatch:
+    inputs:
+      target_branch:
+        description: >-
+          JSON array of branches to target. The first branch in the array will
+          be marked as the latest release.
+        required: false
+        type: string
+        default: '["release/0.6","release/0.5"]'
```
[the remark this answers is not in the document]
Comment (inahga) [reply .github/workflows/make-release.yml:12]: Yep. When we're ready to promote it to a production-ready release, we can just run the workflow normally, but in the mean time we'll have to do it manually.

 The main blocker is that `cargo-edit` doesn't [really support incrementing prerelease versions](https://github.com/killercup/cargo-edit/issues/700). (FWIW I'm not sure if the available alternatives do, like `release-plz`.)

File: .github/workflows/make-release.yml (hunk 1 above)
Comment (divergentdave) [reply .github/workflows/make-release.yml:40]: We could write the version of cargo-edit into the workflow, pass it to cargo install, and put it in the cache key, as we do with a couple other Rust utilities. I expect the main reason we'll want to upgrade cargo-edit is to keep up with Cargo manifest format additions.

Comment (inahga) [reply .github/workflows/make-release.yml:40]: It does take about 3 minutes on a runner, which is an annoying amount of time when iterating on the action, but I suppose not a big deal given the whole release CI will take about an hour.

Either way I was figuring we can just do `gh cache delete Linux-cargo-edit` when we want to flush the cache and get a new version.

Comment (inahga) [reply .github/workflows/make-release.yml:40]: That's a good idea, I see we do that in https://github.com/divviup/libprio-rs/blob/main/.github/workflows/supply-chain.yml. I'll adopt this.

Comment (branlwyd) [reply .github/workflows/make-release.yml:40]: Hm, yeah, I think I like the idea of fixing the version of `cargo-edit` (following the more general principle that the versions of dependencies should be fixed).

**Pull request merged**